### PR TITLE
Jonmv/http url in configserver client

### PR DESCRIPTION
**Pull request by jonmv**
@bratseth please review and merge.

### CODE REVIEW

File: vespajlib/src/main/java/ai/vespa/http/DomainName.java [hunk 1]
```diff
@@ -15,8 +15,8 @@
  */
 public class DomainName extends PatternedStringWrapper<DomainName> {
 
-    static final Pattern labelPattern = Pattern.compile("([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9-]{0,61}[A-Za-z0-9])");
-    static final Pattern domainNamePattern = Pattern.compile("(" + labelPattern + "\\.)*" + labelPattern);
+    public static final Pattern labelPattern = Pattern.compile("([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9-]{0,61}[A-Za-z0-9])");
```
Comment (bratseth) [inline vespajlib/src/main/java/ai/vespa/http/DomainName.java:18]: Doesn't look like these need to be public?

**Review by bratseth — APPROVED** [review]
poetic and/or correct

**Comment by bratseth** [comment]
HttpConfigServerClientTest.testRetries:61 Unexpected exception type thrown ==> expected: <java.io.UncheckedIOException> but was: <java.lang.IllegalArgumentException>

File: vespajlib/src/main/java/ai/vespa/http/DomainName.java (hunk 1 above)
Comment (jonmv) [reply vespajlib/src/main/java/ai/vespa/http/DomainName.java:18]: Not the label. The domain pattern is used by `HostName`. It's probably better to just replicate the pattern. Will do that.

**Pull request merged**